MultiServer.py

```diff
@@ -973,7 +976,10 @@ def register_location_checks(ctx: Context, team: int, slot: int, locations: typi
             "hint_points": get_slot_points(ctx, team, slot),
             "checked_locations": new_locations,  # send back new checks only
         }])
-
+        old_hints = ctx.hints[team, slot].copy()
+        ctx.recheck_hints(team, slot)
+        if old_hints != ctx.hints[team, slot]:
```

Comment:
Wouldn't it make sense to use our helper here?
```suggestion
        if old_hints != ctx.get_rechecked_hints(team, slot):
```

Reply:
Style wise, I don't like "hiding" mutations in an if statement.

Reply:
should we get rid of the helper then? any use of the helper will hide mutation

Reply:
Or potentially rename the helper to be more clear what it does.

Reply:
its only other use currently is outside an if statement.